src/planwise/component/engine.clj

```diff
@@ -164,7 +170,8 @@
                                   (assoc provider :unsatisfied total-demand)))
                               (:providers result-step1))]
 
-    (let [updated-sources           (map #(select-keys % [:id :quantity]) (:sources result-step1)) ; persist only id and quantity
+    (let [initial-quantities        (reduce (fn [tree {:keys [id quantity] :as source}] (assoc tree (keyword (str id)) quantity)) {} sources)
```

Comment:
You can use the `id` as-is for keying the sources map. There's no need to turn it into a keyword.

Reply:
Also, this should be formatted into multiple lines for readability.